File: src/client/AM.PA.MonitoringTool/Shared/IVisualization.cs
// Created by André Meyer (t-anmeye@microsoft.com) working as an Intern at Microsoft Research
// Created: 2015-11-24
// 
// Licensed under the MIT License.

using Shared.Helpers;

namespace Shared
{
    /// <summary>
    /// Currently, only a daily and weekly retrospection is supported.
    /// a monthly retrospection will be added in the future
    /// </summary>
    public enum VisType
    {
        Day,
        Week
    }

    /// <summary>
    /// These are also the sized defined in the CSS
    /// </summary>
    public enum VisSize
    {
        Small,
        Square,
        Wide,
        //Large
    }

    public interface IVisualization
    {
        string Title { get; }
        bool IsEnabled { get; set; }
        VisSize Size { get; }
        VisType Type { get; }
        int Order { get; set; }
        string GetHtml();
    }

    public class BaseVisualization
    {
        public string Title { set; get; }
        public bool IsEnabled { get; set; }
        public VisSize Size { get; set; }
        public VisType Type { get; set; }
        public int Order { get; set; }
        public virtual string GetHtml()
        {
            return VisHelper.Error("Not yet implemented");
        }
    }
}

File: src/client/AM.PA.MacOs/PersonalAnalytics/Shared/IVisualization.swift
//
//  Visualization.swift
//  PersonalAnalytics
//
//  Created by Chris Satterfield on 2017-05-27.
//

import Foundation

enum VisConstants{
    static let Week: String = "week"
    static let Day: String = "day"
}

protocol IVisualization{
    
    //size of visualization to be rendered
    //Square...
    var Size:String { get }
    var color:String { get }
    var title: String { get }
    var _type: [String] { get }
    
    init() throws
    
    func getHtml(_ _date: Date, type: String) -> String
    
}
